[core] Expose mbgl::style::Light::setProperty

brunoabinader opened the pull request:
Works the same way as `mbgl::style::set{Paint,Layout}Property` functions.

pozdnyakov commented on src/mbgl/style/light.cpp:
```diff
@@ -24,6 +31,118 @@ Mutable<Light::Impl> Light::mutableImpl() const {
+    };
+
+    MAPBOX_ETERNAL_CONSTEXPR const auto properties = mapbox::eternal::hash_map<mapbox::eternal::string, uint8_t>({
+        { "anchor", static_cast<uint8_t>(Property::Anchor) },
```
`mbgl::underlying_type()`

brunoabinader commented:
@tmpsantos @pozdnyakov thanks for the review. Added light properties test and replaced the static cast with `mbgl::underlying_type` also in the style layer properties template code.

pozdnyakov commented on src/mbgl/style/layers/raster_layer.cpp:
```diff
@@ -282,7 +283,7 @@ TransitionOptions RasterLayer::getRasterSaturationTransition() const {
 using namespace conversion;
 
 optional<Error> RasterLayer::setPaintProperty(const std::string& name, const Convertible& value) {
-    enum class Property : uint8_t {
```
why is that not `uint8_t` any more?
